src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CollectionBackedStoreProvider.cs

```diff
@@ -0,0 +1,159 @@
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal class CollectionBackedStoreProvider : IStorePal
+    {
+        private readonly X509Certificate2Collection _certs;
+
+        internal CollectionBackedStoreProvider(X509Certificate2 cert)
+        {
+            _certs = new X509Certificate2Collection(cert);
+        }
+
+        internal CollectionBackedStoreProvider(X509Certificate2Collection certs)
+        {
+            _certs = new X509Certificate2Collection(certs);
```

Comment:
I'm a little confused about what's supported.  Add throws an InvalidOperationException, and ExportX509Der only uses _certs[0], so I was assuming that this currently only supports wrapping a single certificate.  Is that true?  If so, should we assert that certs.Count <= 1?